### PR TITLE
[Prometheus] Make the base URL configurable

### PR DESCRIPTION
**Pull request by andrewhowdencom**

### CODE REVIEW

**Comment by andrewhowdencom** [comment]
I signed it

**Comment by andrewhowdencom** [comment]
Needs work.

**Comment by andrewhowdencom** [comment]
In theory, adjusted, though the fix is perhaps less nice.

File: stable/prometheus/README.md [hunk 1]
```diff
@@ -49,6 +49,7 @@ Parameter | Description | Default
 `alertmanager.image.repository` | alertmanager container image repository | `prom/alertmanager`
 `alertmanager.image.tag` | alertmanager container image tag | `v0.5.1`
 `alertmanager.image.pullPolicy` | alertmanager container image pull policy | `IfNotPresent`
+`alertmanager.baseURL` | The prefix slug at which the server can be accessed | `/#/status`
```
Comment (mgoodness) [inline stable/prometheus/README.md]: Default `alertmanager.baseURL` should be ``, no?

File: stable/prometheus/values.yaml [hunk 2]
```diff
@@ -301,6 +306,11 @@ server:
   ## only used if alertmanager.enabled = false
   alertmanagerURL: ""
 
+  ## The URL prefix at which the container can be accessed. Useful in the case the '-web.external-url' includes a slug
+  ## so that the various internal URLs are still able to access as they are in the default case.
+  ## (Optional)
+  baseURL: "/status"
```
Comment (mgoodness) [inline stable/prometheus/values.yaml]: Should be ""

File: stable/prometheus/templates/server-deployment.yaml [hunk 3]
```diff
@@ -40,7 +40,7 @@ spec:
           imagePullPolicy: "{{ .Values.server.image.pullPolicy }}"
           args:
           {{- if or .Values.alertmanager.enabled .Values.server.alertmanagerURL }}
-            - --alertmanager.url={{- if .Values.alertmanager.enabled }}http://{{ template "prometheus.alertmanager.fullname" . }}:{{ .Values.alertmanager.service.servicePort }}{{- else }}{{ .Values.server.alertmanagerURL }}{{- end }}
+            - --alertmanager.url={{- if .Values.alertmanager.enabled }}http://{{ template "prometheus.alertmanager.fullname" . }}:{{ .Values.alertmanager.service.servicePort }}{{.Values.alertmanager.baseURL }}{{- else }}{{ .Values.server.alertmanagerURL }}{{- end }}
```
Comment (mgoodness) [inline stable/prometheus/templates/server-deployment.yaml]: Super-nit: `{{.Values.alertmanager.baseURL }}` needs a space before the first period.

**Review by mgoodness — CHANGES_REQUESTED** [review]
Almost there.

File: stable/prometheus/values.yaml (hunk 2 above)
Comment (andrewhowdencom) [reply stable/prometheus/values.yaml]: ooops!

**Comment by andrewhowdencom** [comment]
Heyaa @mgoodness; 

I made the amendments, but squashed them into the last commit (I don't think that commit added anything of value to the patchset, and was defensively part of the second commit). 

Let me know if you need anything else. I haven't tested this just yet, but the changes seemed simple

**Review by mgoodness — APPROVED** [review]

**Pull request merged**